### PR TITLE
[SDK] Document Secrets in YAML Reference

### PR DESCRIPTION
**Pull request by mbdcos**
Update reference docs to reflect new fields.
```
pods:
  pod_name:	
     secrets:
       secret_name1:
           secret: hello-world/secret1              <secret path>
           env-key: SECRET1_ENV                <environment variable name>
       secret_name2:
           secret: hello-world/secret2             <secret path>
           file: some/path/secret2_file             <file path>

```

### CODE REVIEW

**Comment by mbdcos** [comment]
@nickbp  Can you take a look at the yaml-reference.md?

File: docs/pages/yaml-reference.md [hunk 1]
```diff
@@ -82,6 +82,22 @@ This documentation effectively reflects the Java object tree under [RawServiceSp
 
     These values are respectively equivalent to `image-name`, `networks`, and `rlimits` under `container`. In each case, only one of the two may be specified at a time. See above. (TODO(nickbp) remove one of the two duplicates?)
 
+  * `secrets`
+
+    This section list the Secrets that will be made available to the pod. Secret content is exposed as a file or as a environment variable.
```
Comment (nickbp) [inline docs/pages/yaml-reference.md]: Can both `env-key` and `file` be set at the same time?

Comment (mbdcos) [inline docs/pages/yaml-reference.md]: yes

Comment (nickbp) [inline docs/pages/yaml-reference.md]: Yes 'and/or' would work here. Feel free to merge after the change

**Pull request merged**